src/Login.tsx

```diff
@@ -1,9 +1,112 @@
-import React from "react";
+import React, { useState } from "react";
+import { useNavigate } from "react-router";
+// import axios from "axios";
 
-const Login = () => (
-	<div>
-		<h1>Login Page</h1>
-	</div>
-);
+type FormFieldsType = {
+	email: { value: string };
+	password: { value: string };
+} & EventTarget;
+
+const Login = () => {
+	const navigate = useNavigate();
+	const [error, setError] = useState<string>();
+	const [loginStatus, setLoginStatus] = useState(false);
+
+	const isValid = (e: React.FormEvent<HTMLFormElement>): boolean => {
+		const target = e.target as FormFieldsType;
+
+		const emailInput = target.email.value;
+		const passwordInput = target.password.value;
+		if (emailInput === "" || passwordInput === "") {
+			setError("some field are incomplete");
+		} else if (passwordInput.length < 8) {
+			setError("password must be more than 8 characters");
+		} else if (emailInput.includes("@") === false) {
+			setError("Email invalid");
```

Comment:
Maybe we can change setError to be more descriptive of a name such as setErrorMessage? That way we know we can write friendlier error messages like "Please enter a valid email" instead of "Email invalid"

Reply:
this has been updated